### PR TITLE
negar/pre_settlement

### PR DESCRIPTION
**Pull request by negar-binary**
check for settlement as well as expiry

### CODE REVIEW

File: scripts/bin/GenerateStaticData.pm [hunk 1]
```diff
@@ -704,6 +704,7 @@ sub _texts {
         push @texts, localize('Adjusted High Barrier');
         push @texts, localize('Adjusted Low Barrier');
         push @texts, localize('Corporate Action');
+        push @texts, localize('The final settlement price may differ from the Indicative price.');
```
Comment (yngshan) [inline scripts/bin/GenerateStaticData.pm]: "from the indicative price."

Comment (ashkanx) [inline scripts/bin/GenerateStaticData.pm]: @negar-binary don't forget this please

File: scripts/bin/GenerateStaticData.pm [hunk 2]
```diff
@@ -705,6 +705,7 @@ sub _texts {
         push @texts, localize('Adjusted High Barrier');
         push @texts, localize('Adjusted Low Barrier');
         push @texts, localize('Corporate Action');
+        push @texts, localize('The final settlement price may differ from the indicative price.');
```
Comment (negar-binary) [inline scripts/bin/GenerateStaticData.pm]: @ashkanx it's done here, see ^. github keeps the old changes in old conversations now.

File: src/javascript/binary/websocket_pages/user/view_popup/view_popupws.js [hunk 3]
```diff
@@ -234,7 +234,7 @@ var ViewPopupWS = (function() {
             if(contract.entry_spot > 0) {
                 containerSetText('trade_details_entry_spot', contract.entry_spot);
             }
-            containerSetText('trade_details_message', contract.validation_error ? contract.validation_error : corporateActionEvent ? '* ' + page.text.localize('This contract was affected by a Corporate Action event.') : '&nbsp;');
+            containerSetText('trade_details_message', contract.validation_error ? contract.validation_error : corporateActionEvent ? '* ' + page.text.localize('This contract was affected by a Corporate Action event.') : contract.is_expired && !contract.is_after_settlement ? page.text.localize('The final settlement price may differ from the indicative price.') : '&nbsp;');
```
Comment (negar-binary) [inline src/javascript/binary/websocket_pages/user/view_popup/view_popupws.js]: @ashkanx changed in both places ^.

File: scripts/bin/GenerateStaticData.pm (hunk 1 above)
Comment (yngshan) [inline scripts/bin/GenerateStaticData.pm]: @negar-binary , I added this on backend. So you can remove this message from front end

Comment (negar-binary) [inline scripts/bin/GenerateStaticData.pm]: @yngshan this is removed in my latest commit... git keeps the message history of old code in this page, please ignore it

**Pull request merged**